eng/common/scripts/Delete-RemoteBranches.ps1

```diff
@@ -1,38 +1,71 @@
 param(
+  # Please use the RepoOwner/RepoName format: e.g. Azure/azure-sdk-for-java
+  [Parameter(Mandatory = $true)]
+  $RepoNameWithOwner,
+  # Please use the RepoOwner/RepoName format: e.g. Azure/azure-sdk-for-java
   [Parameter(Mandatory = $true)]
   $RepoOwner,
+  # Please use the RepoOwner/RepoName format: e.g. Azure/azure-sdk-for-java
+  [Parameter(Mandatory = $true)]
+  $RepoName,
   # Use this if a pull request might have been opened from one repo against another.
   # E.g Pull request opened from azure-sdk/azure-sdk prBranch --> Azure/azure-sdk baseBranch
   $ForkRepoOwner,
   [Parameter(Mandatory = $true)]
-  $RepoName,
-  [Parameter(Mandatory = $true)]
-  $BranchPrefix,
+  $BranchPrefix,  
+  # Date format: e.g. Tuesday, April 12, 2022 1:36:02 PM. Allow to use other date format.
+  $LastCommitOlderThan=$env:LASTCOMMITDATEOVERRIDE,
   [Parameter(Mandatory = $true)]
   $AuthToken
 )
 
 . (Join-Path $PSScriptRoot common.ps1)
 
-LogDebug "Operating on Repo [ $RepoName ]"
+function GetCommitDate($commitUrl) {
+  try {
+    $commitResponse = Invoke-WebRequest $commitUrl
+    $commitData = ConvertFrom-Json $commitResponse
+    if (!$commitData.committer -or !$commitData.committer.date) {
+      LogDebug "No date returned from the commit sha. Skipping..."
+    }
+    return $commitData.committer.date
+  }
+  catch {
+    LogError "Failed to invoke $commitUrl with exception:`n$_"
+    exit 1
+  }
+}
+
+LogDebug "Operating on Repo [ $Repo ]"
+if (($Repo -split "/").Count -lt 2) {
```

Comment:
I would simply check and if RepoNameWithOwner is not set set it to `$RepoOwner/$RepoName`. Actually we should do something exactly like we already do at https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/scripts/create-tags-and-git-release.ps1#L17. We should just call this RepoId as well as I like that name better then RepoNameWithOwner.